web/src/stream_list_sort.ts

```diff
@@ -23,6 +23,25 @@ export function get_stream_ids(): number[] {
     return [...all_streams];
 }
 
+function current_section_ids_for_streams(): Map<number, StreamListSection> {
+    const map = new Map<number, StreamListSection>();
+    for (const section of current_sections) {
+        for (const stream_id of [
+            ...section.streams,
+            ...section.muted_streams,
+        ]) {
+            map.set(stream_id, section);
+        }
+    }
+    return map;
+}
+
+export function current_section_id_for_stream(stream_id: number): string {
```

Comment:
This function is a bit dangerous for performance; it sounds like it should be `O(1)`, but is actually `O(N)`. 

Probably worth a comment: 

> Warning: This function is O(n), so it should not be called in a loop.

I checked all the existing callers are fine on that concern.